modules/imgproc/src/featureselect.cpp

```diff
@@ -251,13 +254,18 @@ static bool ocl_goodFeaturesToTrack( InputArray _image, OutputArray _corners,
             const Corner & c = corner_ptr[i];
 
             corners.push_back(Point2f((float)c.x, (float)c.y));
+            corners_values.push_back(eig.getMat(ACCESS_READ).at<float>(c.y, c.x));
```

Comment:
> .getMat(ACCESS_READ)

`.getMat()` / `.getUMat()` in a loop is not a good idea

---

Did you run perf tests locally?